src/models/event-timeline-set.ts

```diff
@@ -522,16 +570,40 @@ export class EventTimelineSet extends TypedEventEmitter<EmittedEvents, EventTime
      * Add an event to the end of this live timeline.
      *
      * @param {MatrixEvent} event Event to be added
-     * @param {string?} duplicateStrategy 'ignore' or 'replace'
-     * @param {boolean} fromCache whether the sync response came from cache
+     * @param {IAddLiveEventOptions} options addLiveEvent options
      * @param roomState the state events to reconcile metadata from
      */
     public addLiveEvent(
```

Comment:
It'd be grand to get a test for this overload, they can be easy to break

Reply:
How far do we want to go with these overload tests? I can add a simple test making sure it doesn't throw an error in some specific scenarios.

But if we want to make sure both method signatures behave the same, we will need to run all of the tests against both overloads.

```js
function addLiveEventOverloadWithOptions(event, opts: IAddLiveEventOptions) {
  return room.addLiveEvent(event, opts);
}

function addLiveEventOverloadWithDirectParameters(event, opts: IAddLiveEventOptions) {
  return room.addLiveEvent(event, opts.duplicateStrategy, opts.duplicateStrategy, opts.duplicateStrategy);
}

[addLiveEventOverloadWithOptions, addLiveEventOverloadWithDirectParameters].forEach((addLiveEvent) => {
  it('foo', () => {
     addLiveEvent(event, {
       duplicateStrategy: 'replace',
       timelineWasEmpty: true,
     });
  });
});

```

Reply:
Ideally we'd implement overloads as wrappers, then we could mock the wrapped function and validate that all overloads result in the same call to the wrapped method, but this is not how the SDK does existing overloads so anything is better than no coverage